src/main/java/com/app/bookstore/service/book/BookServiceImpl.java

```diff
@@ -53,4 +57,14 @@ public BookDto updateBookById(Long id, CreateBookRequestDto bookDto) {
         book.setId(id);
         return bookMapper.toDto(bookRepository.save(book));
     }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(Long id) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can't find category by id: " + id));
+        return bookRepository.getBooksByCategoriesContaining(category).stream()

```

Comment:
Is it possible to search by category ID, not by category object itself? It will allow to removal of logic of searching category